src/main.rs

```diff
@@ -160,12 +153,30 @@ fn start_program(sess: &mut Session, elf: &Elf) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+/// Set rtt to blocking mode
+fn set_rtt_to_blocking(
+    core: &mut Core,
+    main_fn_address: u32,
+    rtt_buffer_address: u32,
+) -> anyhow::Result<()> {
+    core.set_hw_breakpoint(main_fn_address)?;
+    core.run()?;
+    core.wait_for_core_halted(Duration::from_secs(5))?;
+
+    const OFFSET: u32 = 44;
+    const FLAG: u32 = 2; // BLOCK_IF_FULL
+    core.write_word_32(rtt_buffer_address + OFFSET, FLAG)?;
```

Comment:
Preexisting issue, but it's odd that this just overwrites any existing flags. `defmt-rtt` initializes them with `NOBLOCK_TRIM`, but this will clear that flag. I would expect us to read-modify-write the flags instead.

Reply:
Sounds reasonable. How would I do that?

Reply:
`read_word_32`, then `| FLAG`, then write that value back